pkg/api/util/doget.go

```diff
@@ -29,13 +29,22 @@ func GetClient(verify bool) *http.Client {
 }
 
 // DoGet is a wrapper around performing HTTP GET requests
+// By default, the underlying connection is kept open after reading the response
 func DoGet(c *http.Client, url string) (body []byte, e error) {
+	return DoGetWithOptions(c, url, false)
+}
+
+// DoGetWithOptions is a wrapper around performing HTTP GET requests with additional options
+func DoGetWithOptions(c *http.Client, url string, close bool) (body []byte, e error) {
```

Comment:
I think the idea here is to avoid a diff to other parts of the agent, but in the end it just adds tech debt here, where the reader is left wondering why there are two DoGet*.  I think it would be better long-term to replace `apiutil.DoGet(.., ..)` everywhere with `apiutil.DoGet(.., .., false)` (or, better yet, `apiutil.DoGet(.., .., apiutil.CloseConnection)`, making the meaning of the third parameter clear to the reader)

Reply:
That's a really good idea! I've just updated the PR with it! Thanks for pointing it out, Dustin!